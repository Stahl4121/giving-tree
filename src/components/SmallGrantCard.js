import React, { useEffect } from 'react';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import firebase from '../firebase.js';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';
import { Link } from 'react-router-dom';
import Text from './Text.js';
import ProgressBar from './ProgressBar.js';
import { makeStyles } from '@material-ui/styles';
import { useDownloadURL } from 'react-firebase-hooks/storage';

const useStyles = makeStyles(theme => ({
  card: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
  },
  cardMedia: {
    height: 250,
  },
  cardContent: {
    flexGrow: 1,
  },
}))

export default function SmallGrantCard(props) {
  const classes = useStyles();

  // Grant details
  const [grantId, setGrantId] = React.useState(props.id);
  const [grant, setGrant] = React.useState(props.grant);
  const [foundation, setFoundation] = React.useState(props.foundation);
  const [nonprofit, setNonprofit] = React.useState(props.nonprofit);
  const [goal, setGoal] = React.useState(props.goal);
  const [raised, setRaised] = React.useState(props.raised);
  const [img, setImg] = React.useState(props.img);
  const [url, setUrl] = React.useState('');

  // Observe grant details
  useEffect(() => { setGrantId(props.grantId); }, [props.grantId]);
  useEffect(() => { setGrant(props.grant); }, [props.grant]);
  useEffect(() => { setFoundation(props.foundation); }, [props.foundation]);
  useEffect(() => { setNonprofit(props.nonprofit); }, [props.nonprofit]);
  useEffect(() => { setGoal(props.goal); }, [props.goal]);
  useEffect(() => { setRaised(props.raised); }, [props.raised]);
  useEffect(() => { setImg(props.img) }, [props.img]);

  // Create reference to firebase storage
  let storage = firebase.storage();
  let storageRef = storage.ref();

  // Get image URL
  const [downloadUrl, loading, error] = useDownloadURL(storageRef.child(img));

  return (
    <Grid item xs={12} sm={6} md={4}>
      <Card className={classes.card}>
        <CardActionArea
          component={Link}
          to={{
            pathname: '/grants/' + grant.split(' ').join('-'),
            state: { grantId: grantId }
          }}>
          <CardMedia
            className={classes.cardMedia}
            image={downloadUrl}
            title="Grant Image"
          />
          <CardContent className={classes.cardContent}>
            <Text type='card-aboveheading' text={nonprofit} />
            <Text type='card-heading' text={grant} />
            <Text type='card-subheading' text={foundation} />
            <ProgressBar goal={goal} raised={raised} />
          </CardContent>
        </CardActionArea>
      </Card>
    </Grid >
  );
}
