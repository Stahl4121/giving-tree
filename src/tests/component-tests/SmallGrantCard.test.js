import '@testing-library/jest-dom'
import React from 'react'
import { shallow } from 'enzyme';
import FirestoreMock from '../firestore.mock'

import SmallGrantCard from '../../components/SmallGrantCard'

const mockFirestore = new FirestoreMock();

const routeComponentPropsMock = {
  history: {},
  location: {},
  match: {}
}


jest.mock('firebase/app', () => ({
  __esModule: true,
  initializeApp: () => { },
  apps: {
    length: true,
  },
  app: () => {
    return {
      auth: () => { },
      firestore: () => { return mockFirestore },
      storage: () => {
        return {
          get: jest.fn(),
          ref: () => {
            return {
              child: jest.fn(),
            }
          }
        }
      },
    }
  },
}));

describe('Small Grant Card basics', () => {

  it('Renders correctly', () => {
    const wrap = shallow(<SmallGrantCard  {...routeComponentPropsMock} />)

    expect(wrap).toMatchSnapshot();
  });

})

