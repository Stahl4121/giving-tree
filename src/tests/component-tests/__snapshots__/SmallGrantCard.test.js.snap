// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Small Grant Card basics Renders correctly 1`] = `ShallowWrapper {}`;
